src/note_compiler.h

```diff
@@ -17,8 +17,14 @@ typedef struct NoteCompiler NoteCompiler;
 
 // --------
 
+typedef enum NoteCompilerState {
+	NOTE_COMPILER_STATE_EXPECTING_NOTE = 0,
+	NUM_NOTE_COMPILER_STATES,
+} NoteCompilerState;
+
```

Comment:
We don't really make use of that atm, but it may help keep it more extensible, so we can keep it for now.